fedimint-bitcoind/src/bitcoincore_rpc.rs

```diff
@@ -162,3 +180,76 @@ where
         })
     }
 }
+
+pub struct ElectrumClient(electrum_client::Client);
+
+impl ElectrumClient {
+    fn new(url: &Url) -> anyhow::Result<Self> {
+        Ok(Self(electrum_client::Client::new(url.as_str())?))
+    }
+}
+
+impl fmt::Debug for ElectrumClient {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("ElectrumClient")
+    }
+}
+use bitcoin_hashes::hex::ToHex;
+use electrum_client::ElectrumApi;
+
+#[async_trait]
+impl IBitcoindRpc for ElectrumClient {
+    async fn get_network(&self) -> Result<Network> {
+        let resp = fedimint_api::task::block_in_place(|| {
+            self.0.server_features().map_err(anyhow::Error::from)
+        })?;
+        Ok(match resp.genesis_hash.to_hex().as_str() {
+            "000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f" => Network::Bitcoin,
+            n => panic!("Unknown Genesis Hash \"{}\"", n),
+        })
+    }
+
+    async fn get_block_height(&self) -> Result<u64> {
+        fedimint_api::task::block_in_place(|| {
+            Ok(self.0.block_headers_subscribe_raw()?.height as u64)
+        })
+    }
+
+    async fn get_block_hash(&self, height: u64) -> Result<BlockHash> {
+        fedimint_api::task::block_in_place(|| {
+            Ok(self
+                .0
+                .block_headers(height as usize, 1)?
+                .headers
+                .get(0)
+                .ok_or_else(|| format_err!("empty block headers response"))?
+                .block_hash())
+        })
+    }
+
+    async fn get_block(&self, _hash: &BlockHash) -> Result<Block> {
+        // TODO: electrum does not seem to be able to return whole blocks
+        // do we **really** need it?
+        unimplemented!()
```

Comment:
This is a blocker. Seems like we're using it for checking peg-out tx status?

What if we checked instead the history of the status of the output? https://docs.rs/electrum-client/0.12.0/electrum_client/trait.ElectrumApi.html#tymethod.script_get_history

Reply:
This is kinda problematic: when targeting core we should not assume an address index or something like this, while electrum apparently can't serve whole blocks. Maybe we can re-define the API boundary? 

Caution: When implementing using e.g. `script_get_history` we still need to wait till consensus height reaches the block height of the returned UTXOs.

Reply:
Maybe Electrum is the wrong way to go and esplora would be much easier? Their API also has an easy way to get a txoutproof and not just an electrum merkleproof (they can probably be converted, but the tree algos necessary are a bit annoying).

Reply:
I guess it's not really a blocker. Just a missing feature. :D

Reply:
How so? Without it we won't recognize change we send to ourselves. Only workaround seems defining an API both core and electrum client can implement (probably with quite some complexity in these impls due to mismatching APIs).

Reply:
I have fixed it now, but I was thinking that - it will just crash on peg-out, so it "supports only federations that never peg-out". :D

Reply:
"federations that never peg-out" sounds like "write only memory" :laughing: